server/bleep/src/agent/tools/proc.rs

```diff
@@ -30,21 +27,36 @@ impl Agent {
         }))
         .await?;
 
+        let relative_paths = paths.iter().map(|p| p.path.clone()).collect::<Vec<_>>();
+        // not sure we need this?
+        // let repos = paths.iter().map(|p| p.repo.clone()).collect::<Vec<_>>();
+
         let results = self
-            .semantic_search(query.into(), paths.clone(), 20, 0, 0.0, true)
+            .semantic_search(SemanticSearchParams {
+                query: Literal::from(&query.to_string()),
```

Comment:
We'll need to propagate `(repo, path)` pairs to the Qdrant query (to avoid the case where the same relative path is present in multiple repos). Maybe we could simplify this logic by having a `{repo}/{path}` field on each Qdrant payload that we can easily filter against.